detectron2/utils/visualizer.py

```diff
@@ -1228,3 +1232,313 @@ def get_output(self):
+        return super().evaluate()
+
+    def reset(self):
+        self._build_dataset_metadata()
+        self._evalset_table = None
+        self._row_idx = 0
+        self._evalset_table_rows = []
+        return super().reset()
+
+
+
+    def _build_dataset_metadata(self):
+        """
+        Builds parsed metadata lists and mappings dicts to facilitate logging.
+        Builds a list of metadata for each of the validation dataloaders. Expects the metadata to present for ech dataloader or
+        combined as a list of dataloaders. If both are present, the former will be given preferance.
+        Useful for writing labels and captions of predictions.
+
+        E.g.
+        # set the properties separately
+        MetadataCatalog.get("val1").property = ...
+        MetadataCatalog.get("val2").property = ...
+
+        Builds 3 metadata objects for stuff and thing classes:
+        1. self.thing_class_names/self.stuff_class_names -- List[List[str]] of category names
+        2. self.thing_index_to_class/self.stuff_index_to_class-- List[Dict[int, str]]
+        3. self.thing_class_names/self.stuff_class_names -- List[wandb.Classes] for logging images to wandb.Table
+
+        """
+        meta = MetadataCatalog.get(self.dataset_name)
+
+        # Parse thing_classes
+        if hasattr(meta, "thing_classes"): 
+            self.thing_class_names = meta.thing_classes
+
+        wandb_thing_classes = []
+        # NOTE: The classs indeces starts from 1 instead of 0. Treat 0 as void, makes for easier vectorized operations
+        for i, name in enumerate(self.thing_class_names, 1):
+            self.thing_index_to_class[i] = name
+            wandb_thing_classes.append({"id": i, "name": name})
+
+        self._table_thing_classes = self.wandb.Classes(wandb_thing_classes)
+
+        # Parse stuff_classes
+        if hasattr(meta, "stuff_classes"):
+            self.stuff_class_names = meta.stuff_classes
+
+        wandb_stuff_classes = []
+        for i, name in enumerate(self.stuff_class_names):
+            self.stuff_index_to_class[i] = name
+            wandb_stuff_classes.append({"id": i, "name": name})
+
+        self._table_stuff_classes = self.wandb.Classes(wandb_stuff_classes)
+
+
+    def _parse_prediction(self, pred):
+        """
+        Parse prediction of one image and return the primitive martices to plot wandb media files.
+        Moves prediction from GPU to system memory.
+
+        Args:
+            pred (detectron2.structures.instances.Instances): Prediction instance for the image
+            loader_i (int): index of the dataloader being used
+
+        returns:
+            Dict (): parsed predictions
+        """
+        parsed_pred = {}
+        if pred.get("instances") is not None:
+            pred_ins = pred["instances"]
+            parsed_pred["boxes"] = (
+                pred_ins.pred_boxes.tensor.tolist()[:10] if pred_ins.has("pred_boxes") else None
+            )
```

Comment:
Hmm yeah. According to you what would be a good number? I guess by default models output 100 predictions?
I set it to 10 for 2 reasons:
1. 100 predictions seem too much for 1 img in the UI
2. more importantly, gathering large tensors( `comm.gather` ) might slow down. 
what do you think?

Reply:
1. What we typically do in visualization is to show all predictions with confidence score > 0.7. Showing all predictions on one image is indeed very noisy in the UI and 0.7 is an empirical threshold we use. It can certainly be a configurable argument as well.

2. I think speed isn't an important factor here as predictions are pretty small tensors, and there is already a limit on total #images as `self.size`.

Reply:
Interesting. 1. sounds good I'll test with this.
2. I was testing with some cityscapes semantic segmentation tasks and gathering data across GPUs got very slow. And there was a warning from `comm.gather` something like `gathering large data 1.x GB `